Source/mpqapi.cpp

```diff
@@ -103,25 +103,21 @@ void __fastcall mpqapi_update_multi_creation_time(DWORD dwChar)
 
 BOOLEAN __fastcall mpqapi_reg_store_modification_time(char *pbData, int dwLen)
 {
-	int v2;          // ebx
-	char *v3;        // ebp
-	char *v4;        // edi
-	unsigned int v5; // esi
-	char *v6;        // ecx
+	char *pbCurrentData;
+	unsigned int i;
+	char *pbDataToXor;
 
-	v2 = dwLen;
-	v3 = pbData;
-	v4 = pbData;
+	pbCurrentData = pbData;
 	if ((unsigned int)dwLen >= 8) {

```

Comment:
This looks like a decomposed for-loop. Possibly with an intrement of 8 per loop:
```
	for (i = 0; i < dwLen; i += 8)
```

Reply:
Originally I refactored it to a loop similar to your suggestion however the resulting assembly code was not the same, the original asm code has the following:
call <imm_fn>
dec esi
jnz $-0xD
So therefore I changed it back to the way it currently is.